pkg/build/prometheus/metrics_via_collectors.go

```diff
@@ -0,0 +1,159 @@
+package prometheus
+
+import (
+	kselector "k8s.io/apimachinery/pkg/labels"
+	"strings"
+
+	"github.com/golang/glog"
+	buildapi "github.com/openshift/origin/pkg/build/apis/build"
+	internalversion "github.com/openshift/origin/pkg/build/generated/listers/build/internalversion"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	separator                = "_"
+	BuildSubsystem           = "openshift_build"
+	TerminalBuildCount       = "phase_terminal_total"
+	TerminalBuildCountQuery  = BuildSubsystem + separator + TerminalBuildCount
+	ActiveBuildCount         = "phase_active_total"
+	ActiveBuildCountQuery    = BuildSubsystem + separator + ActiveBuildCount
+	BuildDuration            = "duration_seconds"
+	BuildDurationBucketQuery = BuildSubsystem + separator + BuildDuration + separator + "bucket"
+	BuildDurationCountQuery  = BuildSubsystem + separator + BuildDuration + separator + "count"
+	BuildDurationSumQuery    = BuildSubsystem + separator + BuildDuration + separator + "sum"
+	StageDuration            = "stages_duration_seconds"
```

Comment:
yeah, it probably should be stage ... an analogous example from the best practice guide:  `http_request_duration_seconds`